docs/pages/javascript.md

```diff
@@ -63,6 +63,19 @@ var Foundation = require('foundation-sites');
 var $dropdown = new Dropdown('#mydropdown');
 ```
 
+#### Tree Shaking
+
+Many bundlers like Webpack, Rollup or Parcel support tree shaking. It consists of the removing the unused code parts from your codebase or your dependencies. Take a look at these articles to know how it works and how to enable it: [How To Clean Up Your JavaScript Build With Tree Shaking](https://www.engineyard.com/blog/tree-shaking), [Why Webpack 2's Tree Shaking is not as effective as you think](https://advancedweb.hu/2017/02/07/treeshaking/) and [Reduce JavaScript Payloads with Tree Shaking](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/tree-shaking/).
+
+Sadly, the "static analysis" promised by these bundlers to detect unused code in dependencies does not usually work, so we recommend you to manually import the Foundation plugins instead, like the following:
+
+```js 
+import Dropdown from 'foundation-sites/js/foundation.dropdown';
```

Comment:
@ncoden I don't like this way of using foundation...

Is there a way to enable something more user friendly (while still being tree shakable)? Such as
```javascript
import { Reveal } from 'foundation-sites';
```

Reply:
@SassNinja i've also tested this pr (see [here](https://github.com/zurb/foundation-sites/issues/11401#issuecomment-412483242)). Would also prefer if there is a way when it's enough to do something like 
```js
import { Reveal } from 'foundation-sites';

$(document).foundation();
```
But it seems that this is not working, not with the current implementation.

Currently i have to do this to get it working: 
```js
import Foundation from 'foundation-sites/js/foundation';
import Reveal from 'foundation-sites/js/foundation.reveal';

var test = new Foundation.Reveal($('.reveal'));
```

Reply:
Foundation needs more than just the plugin to work properly, if you mean that.

Reply:
@SassNinja Sadly, as explained in the paragraph just above, the following _should work in theory_...
```js
import { Reveal } from 'foundation-sites';
```

...but **in practice** does not, as "pure modules" are not well detected by bundlers, and a direct import is recommended. For example the `lodash` library faced the same problem and they created [babel-plugin-lodash](https://github.com/lodash/babel-plugin-lodash) to convert things like `import map from 'lodash'` to `import _map from 'lodash/map'`.

So at best it depends on the user stack to choose whether he can use the named import and try to keep tree shaking working.

Reply:
Poke @jdalton, It would be nice to get your opinon on this 😉.

Reply:
@marvinhuebner I prefer the `import { Reveal } from 'foundation-sites';` approach too. But I just want to make something that works 😉

Reply:
For side-effect free ability: `lodash-es` is detected as pure by webpack v4 and parcel.js with the `lodash-es`'s use of the package.json field `"sideEffects": false,`.

Reply:
I tested with `"sideEffects": false,` and followed all the ressources I found on tree shaking. I am unable to make this work with imports from `foundation-sites`.

Take a look at https://github.com/ncoden/foundation-sites-tree-shaking. Please tell me if I got something wrong. The project rely on this branch `chore/tree-shaking`, but locally it's linked to my local `foundation-sites` clone.

Reply:
`sideEffects: false` is a per package setting not an application setting. This is why it's set in `lodash-es` because I, as the package author, am stating that its modules are side-effect free.

Reply:
> `sideEffects: false` is a per package setting

Yes I know. I set `sideEffects: false` locally in `foundation-sites`

> but locally it's linked to my local foundation-sites clone with sideEffects set.

Reply:
I'm curious. As Foundation uses jQuery and extends it and also relies on extending global Foundation object I'm not really sure if Foundation can be considered sideEffects free.

Reply:
Foundation plugins can be considered as without side effects, as they must be imported and initialized on a component to work. It's the same for the Foundation Core that is manually added to jQuery `addTojQuery()`. I guess the "no side effects" only means: do not rely on others components without importing them. But Foundation core and plugins import all their dependencies.

Reply:
@jdalton @DaSchTour Would you know what I am missing here (https://github.com/ncoden/foundation-sites-tree-shaking) ? I don't see what I could configure make tree shaking working with imports from `foundation-sites`.

Reply:
@SassNinja @DaSchTour  So what's your opinon on this ?

Reply:
@ncoden I have no idea. I never had any problems with tree shaking but I also haven't used or dependend on any globals for a long time.